src/components/transaction/SolverCompetition/index.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useEffect, useState } from 'react'
+import { EmptyItemWrapper } from 'components/common/StyledUserDetailsTable'
+import CowLoading from 'components/common/CowLoading'
+import { Network } from 'types'
+import { useGetSolverCompetition } from 'hooks/useSolverCompetitionData'
+import { Container, Table, ContentCard } from './styled'
+import { HelpTooltip } from 'components/Tooltip'
+import { RowWithCopyButton } from 'components/common/RowWithCopyButton'
+import { sendEvent } from 'components/analytics'
+import { faCheckCircle, faCircleXmark } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import SolutionsTable from './SolutionsTable'
+import { Order, Solution } from 'api/operator'
+
+import { abbreviateString } from 'utils'
+import { LinkWithPrefixNetwork } from 'components/common/LinkWithPrefixNetwork'
+import ClearingPrices from './ClearingPrices'
+import { Title } from 'apps/explorer/pages/styled'
+import { useTransactionData } from 'hooks/useTransactionData'
+import { useWeb3 } from 'api/web3/hooks'
+
+interface SolverCompetitionParams {
+  txHash: string
+  networkId: Network | undefined
+  orders: Order[] | undefined
+}
+
+interface StatusType {
+  type: string
+}
+
+const StatusIcon = ({ type }: StatusType): JSX.Element => {
+  const icon = type == 'executed' ? faCheckCircle : faCircleXmark
+  const color = type == 'executed' ? '#00D897' : '#F1356E'
+  return <FontAwesomeIcon icon={icon} color={color} />
+}
+
+export function SolverCompetition(params: SolverCompetitionParams): JSX.Element {
+  const { networkId, txHash } = params
+  const web3 = useWeb3()
+  const [currentBlock, setCurrentBlock] = useState<number | undefined>()
+  const { isLoading: isLoadingTransactionData, trace } = useTransactionData(networkId, txHash)
+
+  useEffect(() => {
+    const getCurrentBlock = async (): Promise<void> => {
+      setCurrentBlock(await web3.eth.getBlockNumber())
+    }
+    getCurrentBlock()
+  }, [web3])
+  if (isLoadingTransactionData || !currentBlock) {
```

Comment:
Ubernit:
Please, add a blank line between useEffect() and if() to improve the code readability